src/HaskellWorks/Data/Network/Ip/Ipv6.hs

```diff
@@ -200,3 +200,7 @@ rangeToBlocks r = rangeToBlocksDL r []
 
 blockToRange :: IpBlock Canonical -> Range IpAddress
 blockToRange b = uncurry Range $ bimap firstIpAddress lastIpAddress (b, b)
+
+instance Contains (IpBlock Canonical) where
```

Comment:
IPv6 blocks are always canonical?

Reply:
No, but `lastIpAddress` for some reason requires it for IPv6.